doc/developer/bestPractices/SpellChecking.rst

```diff
@@ -0,0 +1,272 @@
+     LaTeX portion begins.
+
+
+The configuration of the "default set of files and directories" is at
+the top of the ``chplspell`` script, and may be easily altered.
+
+
+Non-interactive invocation
+==========================
+
+The simplest use is to produce a non-interactive report for the
+default files and directories of the Chapel repository.  As of the
+date of this writing, there were still some words yet to be added to
+the project dictionary or corrected, which make a good example:
+
+  .. code-block:: text
+
+     $ chplspell --report-only
+
+     doc/developer/bestPractices/README:50: 'chplspell' not found in dictionary (from token 'chplspell')
+     CHANGES.md:445: 'chpldocumentation' not found in dictionary (from token 'chpldocumentation')
+     CHANGES.md:2001: 'pshm' not found in dictionary (from token 'pshm')
+     CHANGES.md:2341: 'pshm' not found in dictionary (from token 'pshm')
+     CHANGES.md:3360: 'circularities' not found in dictionary (from token 'circularities')
+     spec/Domains.tex:367: 'parsafe' not found in dictionary (from token 'parsafe')
+
+
+``chplspell`` may also be invoked on only particular files or directories
+named on the command line.  For example, this file with one tyop:
+
+  .. code-block:: text
+
+     $ chplspell --report-only doc/developer/bestPractices/SpellChecking.rst
+     doc/developer/bestPractices/SpellChecking.rst:109: 'tyop' not found in dictionary (from token 'tyop')
+     $
+
+(The project dictionary now includes the word "tyop" for this file, so
+the above command no longer produces that result.)
+
+
+Interactive invocation
+======================
+
+``scspell`` provides an interactive mode for making corrections and for
+adding words to the various dictionaries.  This mode is also available
+through ``chplspell``.
+
+See `the scspell page on python.org`__ for details.
+
+``chplspell``'s invocation of ``scspell`` makes any requested
+dictionary changes to ``$CHPL_HOME/.scspell/dictionary``
+
+__ scspell_
+
+
+
+Dictionary file details
+=======================
+
+This section provides a few details about the format of the dictionary
+file.  Understanding of these details is not necessary to make use of
+chplspell as described above.  It will be helpful in making use of the
+more advanced options in the next section.
+
+
+The natural language word list contains the words that may appear in
+any file being spell checked.  It is the last word list in the
+dictionary file, under the heading ``NATURAL:``.
+
+
+A "programming language" word list is used in addition to the natural
+language word list when the file being checked matches one of the file
+extensions given for that word list.  They appear in the dictionary
+file on lines beginning with ``FILETYPE:``, e.g.,
+
+  .. code-block:: text
+
+     FILETYPE: TeX/LaTeX; .tex, .bib
+
+
+A per-file word list is used in addition when a file has a matching
+"file id".  These are stored in the dictionary file under ``FILEID:``
+headings, e.g.
```

Comment:
How does one generate a new file id?  Apologies if this is listed elsewhere, I didn't recall seeing it.